src/core/workflow_engine.py

```
@@ -608,6 +608,59 @@ def _calculate_cleanup_schedule(self, execution_order: List[str]) -> Dict[str, L
+                elif from_output in self.execution_context:
+                    context[to_input] = self.execution_context[from_output]
+
+        # Also include any global context that might be needed
+        for key, value in self.execution_context.items():
+            if key not in context:
+                context[key] = value
+
+        return context
+
+
+    def _build_node_context(self, node_id: str) -> Dict[str, Any]:
```

Flake8 (7.3.0)

[error] redefinition of unused '_build_node_context' from line 611

(F811)